### ISSUE TITLE
an issue with downloads again?

### ISSUE THREAD

**Issue by stilldreame** [d-fi/releases#28]
when the last version has been released, the were no issues with downloads at all. 
But today some releases/tracks are not downloading at all no matter what quality to choose (both deezer and spotify)

(windows)

**Comment by stilldreame** [comment]
upd: nvm. I guess everyone is aware

**Comment by MedoSaeid** [comment]
same issue

**Comment by thegass** [comment]
con confirm. all tracks say "warn Skipped "XXXXXXXX", track not available."

**Comment by Mc-Seam** [comment]
Looks like Deezer has changed something in API. As a result all flac downloads dead, D-FI looks completely broken, some other downloaders still can download MP3....

**Comment by MedoSaeid** [comment]
Use d-fi v2.0.8 to download MP3 - 320kbps, FLAC is dead for now.

**Comment by gcbspender** [comment]
Wasn't able to download 320 with release v2.0.8 so updated to 2.0.10, but same issue.  FLAC nor 320 or 128 download.  All bring up the Skipped error.  I even updated my ARL with my account info and still nothing.  Thanks for the help!

**Comment by richardestes** [comment]
320 kbps isn't working on 2.0.10. Login is broken on 2.0.8 too.

**Comment by stilldreame** [comment]
Well, free-mp3-download.net still downloads flac and mp3 320 tracks available on deezer. The problem is that files it downloads are going without tags
Smloadr still downloads 320 mp3s as well.

So I use them both like this: download any track via smloadr, then download flacs via free-mp3-download.net, and then put flacs in the same folder created by smloadr. Afterwards I use mp3 file as sample to put metadata to flacs via AIMP tag editor (few clicks literally), but track number and track title you'll have to put by yourself. But whatever, it worth it.

**Comment by JuaniElias** [comment]
It's working fine and as expected now with release v2.1.0, at least for me on Windows 10

**Comment by FFSMNRK** [comment]
Not on Mac 10.9 :( I tried downloading a track in FLAC with 2.1.0, no luck. "Account can't stream FLAC", it says.

**Comment by Mc-Seam** [comment]
Not on Linux either (D-FI 2.1.1, Debian testing).
 
 **→ Your account can't stream FLAC tracks**

Tried to use another arl (d-fi -a <<arl here>>). 

Result:

✖ error NEED_USER_AUTH_REQUIRED,Require user auth

Arl is legit.

**Comment by nkawanto** [comment]
Similar issue on Windows, 
 → Your account can't stream FLAC tracks
 
 Tried 320kbps,
 → EACCES: permission denied, open 'd-fi_

**Comment by nkeor** [comment]
For anyone wondering, 2.1.0 needs a HiFi arl in order to download FLACs. This is from Deezer's side, and thus cannot be bypassed.

**Comment by gcbspender** [comment]
Just installed 2.1.1.  I get the error - 'your account can't stream  320 tracks'.  Same thing for FLAC.  but 128 works.

**Comment by nkeor** [comment]
Deezer patched HQ (320kbps), so now with a free account you can only download 128kbps files.

An HQ (premium) account is needed for 320kbps, and a HiFi account is needed for FLAC.

**Comment by corptact** [comment]
any hope of a workaround this patch? it'd be a real tragedy to lose such a powerful tool like this

**Comment by nkeor** [comment]
Unfortunately, as I stated on a previous comment, this is on Deezer's side.

Deezer had an encrypted API endpoint, and we were able to generate the track download URLs.
They now removed that API endpoint, and the only way to get the track URLs is using their official API, which needs the user's token. They don't provide the URLs for users that don't have the 'correct' account (HQ for 320, HiFi for FLAC).

Maybe we figure out how their API generates the track URLs, but I doubt it. d-fi has to provide an authentication token for each track request.
